frontends/common/constantFolding.h

```diff
@@ -162,6 +162,21 @@ class ConstantFolding : public PassManager {
         if (typeMap != nullptr) passes.push_back(new ClearTypeMap(typeMap));
         setName("ConstantFolding");
     }
+
+    /// Function definition for hook to filter which Expression objects are processed.
+    ///
+    /// Function should return nullptr to process the Expression, or a non-null pointer
+    /// to bypass processing. Returned value will be returned by the postorder function.
+    using filter_hook_t = std::function<const IR::Expression *(Visitor *, const IR::Expression *)>;
```

Comment:
I think it would be better to name the type `FilterHook` to match the naming conventions.